staging/src/k8s.io/kube-aggregator/pkg/apiserver/apiserver.go

```diff
@@ -300,7 +300,7 @@ func (c completedConfig) NewWithDelegate(delegationTarget genericapiserver.Deleg
 			ctx, cancel := context.WithCancel(context.Background())
 			go func() {
 				select {
-				case <-postStartHookContext.StopCh:
+				case <-postStartHookContext.Done():
```

Comment:
@sttts  do you know what is happening here? L298-L308. We basically constructed a new background ctx and used it, over post start one. This does not look right, but not sure I want to change it now.

Reply:
good question. I don't understand the comment either.

Reply:
actually, the comment is the same in the other cases. Here, the helper is not used. Maybe just an oversight at the time the helper was introduced. Worth to check the PR when it was added.

Reply:
So, I think it is safe to change this one too.

Reply:
This looks like leftover code: `postStartHookContext` used to have a stop channel. This function here then constructed a context from that (it should have done that with `wait.ContextForChannel`, but that's a different story).

Now `postStartHookContext` itself is a proper context.

So removing `ctx` and replacing it with `postStartHookContext` should be okay.

Reply:
ack. Will make the change too. It looked strange. So was thinking if we have cases where ctx gets terminated and we need it not to be or smth similar.